src/ui/ui_exports.js

```diff
@@ -37,22 +43,43 @@ class UiExports {
     this.consumers.push(consumer);
   }
 
+  // when module ids are defined in a uiExports spec they can
+  // be defined as relative to the public directory
+  resolveModulePath(plugin, moduleId) {
+    const isRelative = moduleId.startsWith('./') || moduleId.startsWith('.\\');
+    if (isRelative) return join(`plugins/${plugin.id}`, moduleId);
+    return moduleId;
+  }
+
+  extendAliases(plugin, alias, idOrIds) {
+    const existingModuleIds = this.aliases[alias] || [];
+    const resolve = id => this.resolveModulePath(plugin, id);
+    const resolvedModuleIds = arr(idOrIds).map(resolve);
+
+    this.aliases[alias] = _.union(existingModuleIds, resolvedModuleIds);
+  }
+
   exportConsumer(type) {
-    for (let consumer of this.consumers) {
+    for (const consumer of this.consumers) {
       if (!consumer.exportConsumer) continue;
-      let fn = consumer.exportConsumer(type);
+      const fn = consumer.exportConsumer(type);
       if (fn) return fn;
     }
 
     switch (type) {
-      case 'app':
       case 'apps':
-        return (plugin, specs) => {
-          for (let spec of [].concat(specs || [])) {
-            let app = this.apps.new(_.defaults({}, spec, {
+        return (plugin, spec) => {
+          for (const rawSpec of arr(spec)) {
```

Comment:
I'm not sure I understand the concern about complexity. The ability to send either one thing or an array of things is pretty darn easy to wrap your head around. In theory, I agree that complexity should be avoided, but I think what we get for this tiny bit of complexity is worth while. That said, I'm probably the worst judge of complexity...

I just really don't want to:
 - force every plugin author to update their uiExports to `apps: [ { ... } ]`
 - prevent plugins from exporting multiple apps
 - create two handlers that do the same thing but respond `app` without iteration and `apps` with iteration.